### DISCUSSION TITLE
why we use y_pred = torch.softmax(y_logits,dim=1).argmax(dim=1) what is it's purpose

### DISCUSSION THREAD

**Discussion by mosaabthemuslim** [mrdbourke/pytorch-deep-learning#875] [Q&A] [answered]

**Comment by LuluW8071** [comment, marked as answer]
```python
y_pred = torch.softmax(y_logits,dim=1).argmax(dim=1) 
```

`Softmax` purpose is to convert all raw logits into probabilities within the range [0, 1] and 
`argmax` purpose is to select the index of the highest probability. 

Therefore, obtaining the index of the highest probability within the range of [0, 1] provides us with the class label of the particular object we are trying to predict.
